src/PackageUrl.cs

```diff
@@ -255,30 +255,31 @@ private static string ValidateType(string type)
             return type.ToLower();
         }
 
-        private static string ValidateNamespace(string @namespace)
+        private string ValidateNamespace(string @namespace)
         {
             if (@namespace == null)
             {
                 return null;
             }
-            return WebUtility.UrlDecode(@namespace.ToLower());
+            return Type switch
+            {
+                "vsm" or "cran" => WebUtility.UrlDecode(@namespace),
+                _ => WebUtility.UrlDecode(@namespace.ToLower())
+            };
         }
 
         private string ValidateName(string name)
         {
             if (name == null)
             {
-                throw new MalformedPackageUrlException("The PackageURL name specified is invalid");
```

Comment:
We should continue to throw this exception as package name and type are mandatory.

Reply:
Restored.